### PR TITLE
Convert guest implementations to use serializable container for load/save

### PR DESCRIPTION
**Pull request by happz**
Instead of custom load/save methods, use those based on dataclasses. We
gain type annotations, automagic conversion from/to YAML, no need for
explicit moves of fields between instances and serialized data.

### CODE REVIEW

**Comment by thrix** [comment]
@happz this looks great :rocket:

**Comment by thrix** [comment]
@happz seems to me this breaks multihost? tests show something

**Comment by thrix** [comment]
> @happz seems to me this breaks multihost? tests show something

nm, it needs another change ...

File: tmt/steps/provision/testcloud.py [hunk 1]
```diff
@@ -215,26 +241,17 @@ def options(cls, how=None):
...
 
     def wake(self, keys=None, data=None):
         """ Wake up the plugin, process data, apply options """
         super().wake(keys=keys, data=data)
 
         # Convert memory and disk to integers
-        for key in ['memory', 'disk']:
-            if isinstance(self.get(key), str):
-                self.data[key] = int(self.data[key])
+        # TODO: can they ever *not* be integers at this point?
```
Comment (thrix) [inline tmt/steps/provision/testcloud.py:251]: @happz seems the only leftover item from this MR, it looks clean to me

**Comment by thrix** [comment]
@happz seems rebase is needed, wanted to do it myself, but was not sure, so rather leave it to you. Seems there were some conflicting changes in the meantime

**Comment by happz** [comment]
Rebased & running all tests

File: tmt/steps/provision/testcloud.py (hunk 1 above)
Comment (happz) [reply tmt/steps/provision/testcloud.py:267]: I'd leave this for later, it's not important enough.

**Review by thrix — APPROVED** [review]
LGTM

**Comment by thrix** [comment]
@happz so how did the full test suite run? Is it ok?

**Comment by happz** [comment]
>

**Comment by happz** [comment]
> @happz so how did the full test suite run? Is it ok?

Yes, all tests did pass. I was a bit surprised...

**Comment by thrix** [comment]
squashed, preparing for merge

**Pull request merged**